src/Tempest.php

```diff
@@ -4,73 +4,79 @@
 
 namespace Tempest;
 
-use Closure;
 use Dotenv\Dotenv;
 use Tempest\Application\Application;
 use Tempest\Application\ConsoleApplication;
 use Tempest\Application\Environment;
 use Tempest\Application\HttpApplication;
 use Tempest\Application\Kernel;
+use Tempest\Bootstraps\ConfigBootstrap;
+use Tempest\Bootstraps\DiscoveryBootstrap;
 use Tempest\Exceptions\ExceptionHandler;
 
 final readonly class Tempest
 {
     private function __construct(
         private Kernel $kernel,
-        private AppConfig $appConfig,
     ) {
     }
 
-    public static function boot(string $root, ?Closure $createAppConfig = null): self
+    public static function boot(string $root): self
     {
         $dotenv = Dotenv::createUnsafeImmutable($root);
         $dotenv->safeLoad();
 
-        $createAppConfig ??= fn () => new AppConfig(
+        $createAppConfig = static fn (): AppConfig => new AppConfig(
```

Comment:
What's the point of this refactor? Why can we no longer pass in an app config?

Reply:
Part of @brendt's response in the issue agreed this isn't necessary as a parameter. My initial suggestion was to replace it with the array of Bootstrap classes that can be added to those we have internally, but he recommended we wait on that for now.